File: app/src/main/java/rraya/nearsoft/com/timesheetsapp/data/IRepository.java
package rraya.nearsoft.com.timesheetsapp.data;

import io.reactivex.Single;

/**
 * Created by oaguilar on 12/19/17.
 */

public interface IRepository {
    Single<String> getTimesheetsTokenFromGoogleToken(String googleToken) throws Throwable;
}

File: app/src/main/java/rraya/nearsoft/com/timesheetsapp/data/IRepository.kt
package rraya.nearsoft.com.timesheetsapp.data

import io.reactivex.Single
import rraya.nearsoft.com.timesheetsapp.data.models.Day

/**
 * Created by oaguilar on 12/19/17.
 */

interface IRepository {
    @Throws(Throwable::class)
    fun getTimesheetsTokenFromGoogleToken(googleToken: String): Single<String>

    @Throws(Throwable::class)
    fun getWeekDaysForWeekStarting(currentDay: String): List<Day>
}
